### PR TITLE
Implements xADGroup Membership support

### PR DESCRIPTION
**Pull request by iainbrighton**
* Adds Members, MembersToInclude, MembersToExclude and MembershipAttribute properties
* Adds ManagedBy and Notes properties
* Adds test coverage
* Formatting updates inline with defined guidelines
* Updates readme.md

### CODE REVIEW

File: Tests/xADGroup.Tests.ps1 [hunk 1]
```diff
@@ -37,159 +31,589 @@ Describe "xADGroup" {
...
                 Mock Remove-ADGroup { }
+            
                 Set-TargetResource @testAbsentParams;
+            
                 Assert-MockCalled Remove-ADGroup -Scope It;
             }
         } #end context Validate Set-TargetResource method
     
     } #end InModuleScope
+
```
Comment (TravisEz13) [inline Tests/xADGroup.Tests.ps1]: Style Nit pick: shouldn't have a newline between two closing curly braces.  This is not blocking.

**Comment by TravisEz13** [comment]
General comment, is the MembershipAttribute property needed.  Can the type be determined from the format of the input?

**Comment by iainbrighton** [comment]
@TravisEz13 Generally people will specify the sAMAccountName property, which is why I've left it as default. We could take an educated guess at the MembershipAttribute if it's not supplied. However, I think it's prudent to assume anything! Thoughts?

**Comment by TravisEz13** [comment]
Agreed, I think we can implement auto-detection of format as a later feature after I thought about it.

**Pull request merged**